### ISSUE TITLE
Misleading exception when `configure` is called on a frozen config 

### ISSUE THREAD

**Issue by solnic** [dry-rb/dry-configurable#41]
I'm getting a weird exception from dry-web/system with 0.7.0:

```
/Users/solnic/.gem/ruby/2.4.1/gems/dry-configurable-0.7.0/lib/dry/configurable.rb:167:in `raise_frozen_config': Cannot modify frozen config (Dry::Configurable::FrozenConfig)
        from /Users/solnic/.gem/ruby/2.4.1/gems/dry-configurable-0.7.0/lib/dry/configurable.rb:69:in `configure'
        from /Users/solnic/.gem/ruby/2.4.1/gems/dry-system-0.7.3/lib/dry/system/container.rb:99:in `configure'
        from /Users/solnic/.gem/ruby/2.4.1/gems/dry-web-0.7.1/lib/dry/web/container.rb:15:in `configure'
```

It doesn't modify any strings, it just calls `configure` for the second time.

**Comment by AMHOL** [comment]
@solnic that appears to be from https://github.com/dry-rb/dry-configurable/blob/e128bdc5f8f4eb0fe52e0dc876e9390049f028fc/lib/dry/configurable.rb#L69, this behavior seems correct to me as `configure` should not be called again after `finalize!`, however, if it's causing issues it can be removed and we can just raise when config is mutated?

**Comment by solnic** [comment]
It'd be good to reconsider freezing config inside `configure` block, it makes various scenarios hard, esp in dry-system where `configure` is called within an inheritance hierarchy, so we hit cases where we still want to configure something, but it's already frozen (which is exactly what happened in my case).

**Comment by zaidan** [comment]
This also breaks one of my use cases, had to revert to 0.6.2.

**Comment by GustavoCaso** [comment]
This issue has been addressed with 1caa28e3b405db806aaf1dc09748e1213ec9bd08. Now you need to explicitly call`finalize!` after you have `configure` your configuration class
